### DISCUSSION TITLE
[SUGGESTION] Shields rework

### DISCUSSION THREAD

**Discussion by matejkar** [tsunamayo/Starship-EVO#4026] [Suggestions]
Currently in the game shields work in a way i haven't seen anywhere else, being games, shows, or movies.
In game the shields recharge after depleted in a matter of seconds, now i know the game was not made specifically for large ship engagements, so i get it that for smaller craft that fast shields recharge does not make a difference. However, lets say we have 2 ships about 100m long, with 50k shield strength both. Now if one ship has more firepower and deals lets say 4k dmg per second/volley, it would take a long amount of time to kill a ship. Now add the shields to that and we have situations where 2 ships are killing each other for 15 minutes, and i don't think anyone would be enjoying sitting there so long (i speak of experience, since i primarly build capitals and saw a lot of lengthy duels which were just boring after some time).

Now my suggestion is to make the shields work like in star wars, ie. they have a certain strength, and after depleted, the shields recharge only when the ship is not taking damage. I would really like to see this in game, because i have planned some kind of engagements between some players i know in ships we built for our fictional universe (i explained that already, not gonna go into more detail, if anyone wants to know more, let me know and i can do an entire issue about it), mostly capitals. And we do not plan to sit there for 30 mins with 3 ships shooting at each other.

So if (when) the dev sees this, i really hope we can get shields working in this way, the shield value can stay as it is, just make the shields recharge only when the ship is not taking damage (lets say about 5-10sec some kind of base value), and is stopped when the ship takes damage, and the shield strength stays at the point where it had time to recharge.

**Comment by Garrett-C** [comment]
Doesn't a system like that discourage slow high damage weapons in favour of faster low damage ones? Because a fast one would keep the shield down while the slow one likely would never hit when the shield was down. 

Some way to use weapons like ion weapons may help to diversify weapons used while providing a faster way to take shields down. Then if shield recharge was just slowed a bit you could find a balance. That said it does result in capacitors feeling even less useful for shields so they would probably need reworked as well.

**Comment by TIKIRobo** [comment]
I feel like what would be better is a total combat refactor and/or upgrade, beams do great damage against shields but it would be interesting and cooler to see say EMP missiles or different types of shields like a type of shield that has more health but can only recharge if the ship hasn't been damaged for x amount of seconds

**Comment by matejkar** [comment]
@Garrett-C as the matter of fact, the larger weapons simply do far too much damage to be ignored, while smaller weapons (lets say small lasers and beams up to L2 large lasers and beams) simply don't do that much damage, and while they seem like the better option, in combat against larger ships, they are actually a drawback, even if the large weapons take from 2 to 5 secs to recharge.

**Comment by matejkar** [comment]
@TIKIRobo well ofc a total combat rework would be way better, i just pointed out one of my biggest issues

**Comment by Garrett-C** [comment]
> @Garrett-C as the matter of fact, the larger weapons simply do far too much damage to be ignored, while smaller weapons (lets say small lasers and beams up to L2 large lasers and beams) simply don't do that much damage, and while they seem like the better option, in combat against larger ships, they are actually a drawback, even if the large weapons take from 2 to 5 secs to recharge.

At the moment yes. But if shields only regen when you aren't taking damage then a constant source of damage would mean no shield regen while a large slow source of damage would likely only ever hit a shield because it would start to regen before a shot can be fired.

**Comment by matejkar** [comment]
> > @Garrett-C as the matter of fact, the larger weapons simply do far too much damage to be ignored, while smaller weapons (lets say small lasers and beams up to L2 large lasers and beams) simply don't do that much damage, and while they seem like the better option, in combat against larger ships, they are actually a drawback, even if the large weapons take from 2 to 5 secs to recharge.
> 
> At the moment yes. But if shields only regen when you aren't taking damage then a constant source of damage would mean no shield regen while a large slow source of damage would likely only ever hit a shield because it would start to regen before a shot can be fired.

Well that encourages people to mount smaller weapons alongside larger ones as well. So i don't really see that much of a problem here.

**Comment by Garrett-C** [comment]
> > > @Garrett-C as the matter of fact, the larger weapons simply do far too much damage to be ignored, while smaller weapons (lets say small lasers and beams up to L2 large lasers and beams) simply don't do that much damage, and while they seem like the better option, in combat against larger ships, they are actually a drawback, even if the large weapons take from 2 to 5 secs to recharge.
> > 
> > 
> > At the moment yes. But if shields only regen when you aren't taking damage then a constant source of damage would mean no shield regen while a large slow source of damage would likely only ever hit a shield because it would start to regen before a shot can be fired.
> 
> Well that encourages people to mount smaller weapons alongside larger ones as well. So i don't really see that much of a problem here.

I see it more as encouraging people to just use the gatling barrel instead of any other barrel type.

**Comment by matejkar** [comment]
Yea i kinda see how the freedom of building would be taken away by some kind of meta weapon setups

**Comment by tsunamayo** [comment]
Hi, yeah this could be easily abused with a small weapon pinging constantly the larger ship - which means no shield at all.
I can reduce the shield base capacity again. It recharge super fast because I guess you put no capacitor on it. Put a single shield and like 5 capacitor and tell me how it charges... A low capacity and a high recharge rate is gonna recharge fast.
Then to negate the shield I prefer to go the anti-shield weapon (Ion ect) route.  
And of course I will balance it again, but I kinda gave up of doing it from feedback, I will have to see it for myself (hence it is taking time).
If you want me to see it for myself send me your ship, and post screenshot of the stats screen so I can see how power is balanced. Also capital ship battle is not supposed to be over a 10s I think we all agree on that.

**Comment by matejkar** [comment]
@tsunamayo hi, yes i think you are right with the stuff you pointed out, i did not have any capacitors on my shields.
here are the 2 ships i used to test this out:

[Equalizer - class heavy cruiser mk.2.zip](https://github.com/tsunamayo/Starship-EVO/files/5902523/Equalizer.-.class.heavy.cruiser.mk.2.zip)
[Coercion-Class Cruiser.zip](https://github.com/tsunamayo/Starship-EVO/files/5902693/Coercion-Class.Cruiser.zip)

**Comment by Garrett-C** [comment]
I fee like down the line capacitors may need a rework to incentivise using them. Currently it's sort of pointless to use them because the faster recharge time of just using more shield blocks. Obviously it's not a critical issue now but something to consider later when combat gets a rework.

**Comment by cloudrambler** [comment]
For me the value of capacitors depends on the ship type and combat scenario. For large full combat ships, I agree that capacitors don't add much value as you want resilient shields that charge quickly. But for non-combat ships that don't plan to stay in the fight, they just need enough shields to run away without getting destroyed. In that instance you would get higher shield density for less power consumption by using the maximum number of capacitors and minimum shield generators. You only get one effective shield charge per fight, but it would be five times higher than for the same number of generators, with low power drain.

This could also be a viable strategy for small fighters, which lack the power and thermal capacity to run huge shield banks, and can easily get out of firing range when their shield goes down.

So personally I think there is enough design flexibility and justification for the shields as they are. What may be missing is that grand weapon that can bring shields down with a few hits, but has to recharge for ages and has no gimble etc. That would be a fitting main weapon for destroyers and larger capital ships.

**Comment by PFSCommissar** [comment]
I'd like to see something like how EVE does it, where you can have the option of Active Tanking (weaker shields but fast recharge), or Passive (slow recharge but strong shields). It would be great if shields constantly recharged, even while being hit. 

For example, say the shield has 1000 hp, and recharges at 10 hp/s. So they get hit by a weapon that does 100 damage every 3 seconds. Initial shot and the shield is down to 900. But before the weapon hits again, the shield is up to 930. 

This would require a very large slowing of recharge rates currently in game, as they'd be too fast for this kind of active tanking, but it would present some very interesting options for ship engineering and defense philosophies. You could even add alternative addons to capacitors that increased the recharge rate at the cost of heat or power
